### PR TITLE
add continue in danswer button to slack bot responses

### PR DESCRIPTION
**Pull request by hagen-danswer**
## Description
- there is now an option for a button to continue slack conversations in the danswer web-app

image

## Backporting (check the box to trigger backport action)
Note: You have to check that the action passes, otherwise resolve the conflicts manually and tag the patches.
- [ ] This PR should be backported (make sure to check that the backport attempt succeeds)

### CODE REVIEW

File: backend/danswer/danswerbot/slack/blocks.py [hunk 1]
```diff
@@ -483,3 +507,114 @@ def build_follow_up_resolved_blocks(
         ]
     )
     return [text_block, button_block]
+
+
+def block_builder(
```
Comment (pablonyx) [inline backend/danswer/danswerbot/slack/blocks.py]: Nit: naming

File: backend/danswer/db/chat.py [hunk 2]
```diff
@@ -250,6 +252,43 @@ def create_chat_session(
     return chat_session
 
 
+def duplicate_chat_session_for_user_from_slack(
+    db_session: Session,
```
Comment (pablonyx) [inline backend/danswer/db/chat.py:256]: Comment for this function would be great- can foresee some confusion on this usage

File: web/src/app/chat/ChatPage.tsx [hunk 3]
```diff
@@ -1810,6 +1817,42 @@ export function ChatPage({
     };
   }
 
+  // Add this near the top of the file where other useEffect hooks are
```
Comment (pablonyx) [inline web/src/app/chat/ChatPage.tsx]: Nit

File: web/src/app/chat/lib.tsx [hunk 4]
```diff
@@ -212,7 +212,6 @@ export async function nameChatSession(chatSessionId: string, message: string) {
     body: JSON.stringify({
       chat_session_id: chatSessionId,
       name: null,
-      first_message: message,
```
Comment (pablodanswer) [inline web/src/app/chat/lib.tsx:215]: Is there a reason this is removed?

File: web/src/app/chat/ChatPage.tsx [hunk 5]
```diff
@@ -453,6 +456,7 @@ export function ChatPage({
         }
       }
       setIsFetchingChatMessages(false);
+      console.log("stuff", chatSession);
```
Comment (pablonyx) [inline web/src/app/chat/ChatPage.tsx]: nit

File: web/src/app/chat/ChatPage.tsx [hunk 6]
```diff
@@ -1810,6 +1817,42 @@ export function ChatPage({
...
+        console.error("Error seeding chat from Slack:", error);
+        setPopup({
+          message: "Failed to load chat from Slack",
+          type: "error",
+        });
+      }
+    };
+
+    handleSlackChatRedirect();
+  }, [searchParams, router]); // Add any other dependencies needed
```
Comment (pablonyx) [inline web/src/app/chat/ChatPage.tsx]: nit

**Review by pablodanswer — APPROVED** [review]

**Pull request merged**